### PR TITLE
Model Hamiltonian in terms of spatial-orbitals

### PR DESCRIPTION
**Pull request by gabrielasd**
Filled method `to_spatial` from HamiltonianAPI class to converts from internal (spin-orbital) format to spatial-orbital one. 
Added tentative functions to reduce (or restore) the Hamiltonian terms permutational symmetries: `reduce_sym` and `restore_sym`

### CODE REVIEW

**Comment by RichRick1** [comment]
Looks great to me! @msricher what do you think?

File: upd/Hamiltonian.py [hunk 1]
```diff
@@ -47,7 +48,69 @@ def to_spatial(self, integral: np.ndarray, sym: int, dense: bool):
...
+            elif sym == 2:
+                for i in range(k):
+                    for j in range(k):
+                        spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
+            elif sym == 4:
+                for i in range(k):
+                    for j in range(k):
+                        spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
+                        spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
+            else:
```
Comment (RichRick1) [inline upd/Hamiltonian.py]: I might be wrong, but it seems like the part 
```
for i in range: 
     for j in range:
```
is duplicating in ternary operators.
would it be less code with the same functionality if we use something like this:

```
for i in range(k):
    for j in range(k):
        if sym == 1:
            spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
            spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
            spatial_int[(j,i,i,j)] = integral[(i,i,j,j)]
            spatial_int[(i,j,j,i)] = integral[(i,i,j,j)]   
        elif sym == 2:
            spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
        elif sym == 4:
            spatial_int[(j, i, j, i)] = spatial_int[(i, j, i, j)]
            spatial_int[(j,j,i,i)] = integral[(i,i,j,j)]
```
It seems like it should do the same thing, but using less code
@gabrielasd  what do you think about it?

Comment (gabrielasd) [reply upd/Hamiltonian.py]: @RichRick1 you are right! I'll fix that, thanks :)

**Comment by RichRick1** [comment]
1. Support only Restricted Hamiltonian. 
2. `reduse_sym` convert anything to 1(None) fold symmetry.

**Comment by RichRick1** [comment]
Looks great to me! @msricher @adsrichards what do you think?
Just a minor typo in [this line](https://github.com/gabrielasd/ModelHamiltonian/blob/fccb00628ee0a6cf38c4e472a151037cdc6edaef/upd/Hamiltonian.py#L176)

**Comment by RichRick1** [comment]
Have the `n` parameter at [this line](https://github.com/theochem/ModelHamiltonian/blob/fccb00628ee0a6cf38c4e472a151037cdc6edaef/upd/Hamiltonian.py#L157) specifying the _type of integral_ (one or two body integral)
Also, we assume, that `sym` parameter is an attribute of the class

**Comment by gabrielasd** [comment]
> Have the `n` parameter at [this line](https://github.com/theochem/ModelHamiltonian/blob/fccb00628ee0a6cf38c4e472a151037cdc6edaef/upd/Hamiltonian.py#L157) specifying the _type of integral_ (one or two body integral) Also, we assume, that `sym` parameter is an attribute of the class

Hi @RichRick1 @PaulWAyers @msricher 
for the converting symmetry function ([expand_sym](https://github.com/theochem/ModelHamiltonian/blob/fccb00628ee0a6cf38c4e472a151037cdc6edaef/upd/Hamiltonian.py#L157)) which will take the {1,2}-body Hamiltonian matrix and the _type of integral_ parameters, how can I check that the user didn't make the mistake of passing the two-body integrals as a one-body type? 
e.g do: `expand_sym(sym, integral, nbody)` with integral = two-body integrals matrix, and nbody=1 (one-body type)

In the current internal format the one- and two-body integrals are both 2D matrices, so I can't use the dimensions to distinguish them.
and I don't think the square root of the number of rows would be a definitive criteria either (number of basis could be pair)

**Pull request merged**